From dd6e527f887ae8b81ffc7e72d651b8578fb5287f Mon Sep 17 00:00:00 2001
From: WyattBlue <wyattblue@auto-editor.com>
Date: Fri, 10 Oct 2025 04:45:01 -0400
Subject: [PATCH] Fix #2029

Add new HWDeviceType enums, skip test when version isn't SemVar.
---
 av/about.py          | 2 +-
 av/codec/hwaccel.pyx | 3 +++
 tests/test_packet.py | 7 +++++++
 3 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/av/about.py b/av/about.py
index 5e5e9109a..610c11155 100644
--- a/av/about.py
+++ b/av/about.py
@@ -1 +1 @@
-__version__ = "16.0.0"
+__version__ = "16.0.1"
diff --git a/av/codec/hwaccel.pyx b/av/codec/hwaccel.pyx
index 257e6e7b2..e9a6bd194 100644
--- a/av/codec/hwaccel.pyx
+++ b/av/codec/hwaccel.pyx
@@ -25,6 +25,9 @@ class HWDeviceType(IntEnum):
     mediacodec = lib.AV_HWDEVICE_TYPE_MEDIACODEC
     vulkan = lib.AV_HWDEVICE_TYPE_VULKAN
     d3d12va = lib.AV_HWDEVICE_TYPE_D3D12VA
+    amf = 13  # FFmpeg >=8
+    ohcodec = 14
+    # TODO: When ffmpeg major is changed, check this enum.
 
 class HWConfigMethod(IntEnum):
     none = 0
diff --git a/tests/test_packet.py b/tests/test_packet.py
index d5552238c..b0920ed3f 100644
--- a/tests/test_packet.py
+++ b/tests/test_packet.py
@@ -1,4 +1,5 @@
 from typing import get_args
+from unittest import SkipTest
 
 import av
 
@@ -55,6 +56,12 @@ def test_set_duration(self) -> None:
 class TestPacketSideData:
     def test_data_types(self) -> None:
         dtypes = get_args(av.packet.PktSideDataT)
+
+        if av.ffmpeg_version_info.startswith("n") or av.ffmpeg_version_info.count(
+            "."
+        ) not in (1, 2):
+            raise SkipTest(f"Expect version to be SemVar: {av.ffmpeg_version_info}")
+
         ffmpeg_ver = [int(v) for v in av.ffmpeg_version_info.split(".", 2)[:2]]
         for dtype in dtypes:
             av_enum = av.packet.packet_sidedata_type_from_literal(dtype)
